modules/@apostrophecms/doc/index.js

```diff
@@ -937,6 +937,248 @@ module.exports = {
+          });
+        }
+        if (typeof key !== 'string') {
+          throw new Error('Key must be a string.', {
+            cause: 'invalidArguments'
+          });
+        }
+        const nsKey = `${namespace}:${key}`;
+
+        const existingValue = _.get(doc, metaPathFull) || {};
+        delete existingValue[nsKey];
+        _.set(doc, metaPathFull, existingValue);
+
+        cleanup(doc.aposMeta, 'aposMeta');
+
+        return metaPath;
+
+        function cleanup(object, path) {
+          if (_.isEmpty(object)) {
+            _.unset(object, path);
+            return true;
+          }
+
+          for (const key of Object.keys(object)) {
+            if (key.includes(':')) {
+              return false;
+            }
+            if (!_.isPlainObject(object[key])) {
+              delete object[key];
+              continue;
+            }
+            if (!cleanup(object[key], `${path}.${key}`)) {
+              return false;
+            }
+
+            delete object[key];
+          }
+
+          return true;
+        }
+      },
+      // Get all meta keys for a given field. It has exactly the same signature as
+      // `setMeta` method, except no key/value should be provided.
+      getMetaKeys(doc, namespace, ...pathArgs) {
+        return Object.keys(
+          _.get(
+            doc,
+            `aposMeta.${self.getMetaPath(...pathArgs)}`
+          ) || {}
+        )
+          .filter(key => key.startsWith(`${namespace}:`))
+          .map(key => key.replace(`${namespace}:`, ''));
+      },
+      // Get the meta path for a given field.
+      // Signature:
+      // `apos.doc.getMetaPath([subobject,] ...pathComponents);`
+      // See `setMeta` for more information about `subobject` and `pathComponents` arguments.
+      //
+      // Returns the path to the meta property withouth the namespace and key.
+      // The returned path can be directly used to access or modify the meta property.
+      // It's supported by all meta API methods.
+      //
+      // Example:
+      // ```js
+      // const path = apos.doc.getMetaPath(subobject, 'address', 'city', 'name');
+      // apos.doc.setMeta(doc, ns, path, 'myMetaKey', 'myMetaValue');
+      // apos.doc.getMeta(doc, ns, path, 'myMetaKey');
+      // apos.doc.removeMeta(doc, ns, path, 'myMetaKey');
+      getMetaPath(...pathArgs) {
+        const args = pathArgs
+          .filter(arg => typeof arg !== 'undefined' && arg !== null);
+
+        let subObject;
+        if (_.isPlainObject(args[0])) {
+          subObject = args.shift();
+        }
+
+        if (args.some(arg => typeof arg !== 'string')) {
+          throw new Error('All path components must be strings.', {
+            cause: 'invalidArguments'
+          });
+        }
+        const pathComponents = args.join('.aposMeta.');
+
+        if (!subObject && !pathComponents) {
+          throw new Error('Not providing both subobject and pathComponents is not allowed.', {
```

Comment:
"You must provide at least a `subobject` or at least one `pathComponent` string."